### PR TITLE
fix: remove-unused-components strips remotely referenced components

### LINKED ISSUE #504: --remove-unused-components strips remotely referenced components
I am currently testing --remove-unused-components with a spec that has $refs pointing to an external specification:
```
          properties:
            items:
              type: array
              items:
                $ref: 'https://my-other-api.my-service.io/openapi.yaml#/components/schemas/Order'
```

If I run openapi bundle, I can produce a dereferenced version:
```
openapi bundle -o openapi.combined.yaml --ext yaml openapi.yaml
```
This gives me a file with an Order schema included. However, if I add --remove-unused-components:
```
openapi bundle --remove-unused-components -o openapi.combined.yaml --ext yaml openapi.yaml
```
Then the Order schema gets immediately removed, because it's not detected as used.

### PR DESCRIPTION
**Pull request by andriyl**
## What/Why/How?
Closes #504  
## Check yourself
- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security
- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/core/src/bundle.ts [hunk 1]
```diff
@@ -267,6 +269,17 @@ function makeBundleVisitor(
     };
   }
 
+  function replaceInlineRef(node: OasRef, resolved: ResolveResult<any>, ctx: UserContext) {
```
Comment (RomanHotsiy) [inline packages/core/src/bundle.ts:272]: Rename it to `resolveBundledComponent` plz

File: packages/core/src/bundle.ts [hunk 2]
```diff
@@ -267,6 +269,17 @@ function makeBundleVisitor(
...
+    const newRefId = ctx.location.source.absoluteRef + '::' + node.$ref;
```
Comment (RomanHotsiy) [inline packages/core/src/bundle.ts:273]: Make this a helper function in `resolve.ts` module:

`function makeRefId(absoluteRef, pointer)`

**Review by RomanHotsiy — CHANGES_REQUESTED** [review]
Left a few suggestion.

File: packages/core/src/bundle.ts (hunk 2 above)
Comment (andriyl) [reply packages/core/src/bundle.ts]: done

File: packages/core/src/bundle.ts (hunk 1 above)
Comment (andriyl) [reply packages/core/src/bundle.ts]: done

**Review by RomanHotsiy — APPROVED** [review]

**Pull request merged**